sdks/java/io/azure/src/test/java/org/apache/beam/sdk/io/azure/blobstore/AzfsResourceIdTest.java

```diff
@@ -0,0 +1,311 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.azure.blobstore;
+
+import static org.apache.beam.sdk.io.fs.ResolveOptions.StandardResolveOptions.RESOLVE_DIRECTORY;
+import static org.apache.beam.sdk.io.fs.ResolveOptions.StandardResolveOptions.RESOLVE_FILE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+import org.apache.beam.sdk.io.fs.ResolveOptions;
+import org.apache.beam.sdk.io.fs.ResourceId;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class AzfsResourceIdTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  static final class TestCase {
+
+    final String baseUri;
+    final String relativePath;
+    final ResolveOptions.StandardResolveOptions resolveOptions;
+    final String expectedResult;
+
+    TestCase(
+        String baseUri,
+        String relativePath,
+        ResolveOptions.StandardResolveOptions resolveOptions,
+        String expectedResult) {
+      this.baseUri = baseUri;
+      this.relativePath = relativePath;
+      this.resolveOptions = resolveOptions;
+      this.expectedResult = expectedResult;
+    }
+  }
+
+  // Each test case is an expected URL, then the components used to build it.
+  // Empty components result in a double slash.
+  private static final List<TestCase> PATH_TEST_CASES =
+      Arrays.asList(
+          new TestCase(
+              "azfs://account/container/", "", RESOLVE_DIRECTORY, "azfs://account/container/"),
+          new TestCase(
+              "azfs://account/container", "", RESOLVE_DIRECTORY, "azfs://account/container/"),
+          new TestCase(
+              "azfs://account/container",
+              "path/to/dir",
+              RESOLVE_DIRECTORY,
+              "azfs://account/container/path/to/dir/"),
+          new TestCase(
+              "azfs://account/container",
+              "path/to/object",
+              RESOLVE_FILE,
+              "azfs://account/container/path/to/object"),
+          new TestCase(
+              "azfs://account/container/path/to/dir/",
+              "..",
+              RESOLVE_DIRECTORY,
+              "azfs://account/container/path/to/"));
+
+  @Test
+  public void testResolve() {
+    for (TestCase testCase : PATH_TEST_CASES) {
+      ResourceId resourceId = AzfsResourceId.fromUri(testCase.baseUri);
+      ResourceId resolved = resourceId.resolve(testCase.relativePath, testCase.resolveOptions);
+      assertEquals(testCase.expectedResult, resolved.toString());
+    }
+
+    // Tests for common gcs paths.
```

Comment:
Update comment, heh.
```suggestion
    // Tests for common Azure paths.
```